File: app/lib/views/video_view/components/active_video_chat.dart
class CallRoom {
  int? id;
  String? name;
  String? uuid;
  int? originator;

  CallRoom({
    this.id,
    this.name,
    this.uuid,
    this.originator,
  });

  toMap() {
    return {
      "id": id,
      "name": name,
      "uuid": uuid,
      "originator": originator,
    };
  }

  fromMap(json) {
    id = json["id"];
    name = json["name"];
    uuid = json["uuid"];
    originator = json["originator"];
  }

  static createFromMap(json) {
    var item = CallRoom();
    item.fromMap(json);
    return item;
  }
}

class ActiveVideoChat {
  List<int> recipients;
  CallRoom? callRoom;
  Map<String, dynamic>? originator;
  bool active;

  ActiveVideoChat({
    required this.recipients,
    this.callRoom,
    this.originator,
    this.active = false,
  });

  toMap() {
    return {
      "recipients": recipients,
      "callRoom": callRoom?.toMap(),
      "originator": originator,
      "active": active,
    };
  }

  fromMap(json) {
    recipients = json["recipients"];
    callRoom = CallRoom.createFromMap(json["callRoom"]);
    originator = json["originator"];
    active = json["active"];
  }

  static createFromMap(json) {
    print("creating active video chat $json");
    var item = ActiveVideoChat(
      recipients: List<int>.from(json["recipients"]),
      callRoom: CallRoom.createFromMap(json["callRoom"]),
      originator: Map<String, dynamic>.from(json["originator"]),
      active: json?["active"] ?? false,
    );
    return item;
  }
}

File: app/ios/Runner/video_call/components/ActiveVideoChat.swift


class ActiveVideoChat {
    var recipients: [Int]
    var callRoom: CallRoom?
    var originator: [String: Any]?
    var active: Bool

    init(recipients: [Int], callRoom: CallRoom? = nil, originator: [String: Any]? = nil, active: Bool = false) {
        self.recipients = recipients
        self.callRoom = callRoom
        self.originator = originator
        self.active = active
    }

    func toMap() -> [String: Any?] {
        return ["recipients": recipients, "callRoom": callRoom?.toMap(), "originator": originator, "active": active]
    }

    func fromMap(json: [String: Any]) {
        recipients = json["recipients"] as? [Int] ?? []
        callRoom = CallRoom.createFromMap(json: json["callRoom"] as? [String: Any] ?? [:])
        originator = json["originator"] as? [String: Any]
        active = json["active"] as? Bool ?? false
    }

    static func createFromMap(json: [String: Any]) -> ActiveVideoChat {
        return ActiveVideoChat(
            recipients: json["recipients"] as? [Int] ?? [],
            callRoom: CallRoom.createFromMap(json: json["callRoom"] as? [String: Any] ?? [:]),
            originator: json["originator"] as? [String: Any],
            active: json["active"] as? Bool ?? false
        )
    }
}
